responder/models.py

```diff
@@ -318,9 +308,9 @@ async def body(self):
             content = self.content
             if self.mimetype is not None:
                 headers["Content-Type"] = self.mimetype
-            if self.mimetype == "text/plain" and self.encoding is not None:
-                headers["Encoding"] = self.encoding
-                content = content.encode(self.encoding)
+            if self.mimetype == "text/plain":
+                # Follow StarletteResponse.charset, which is set as "utf-8"
+                content = content.encode(getattr(StarletteResponse, "charset", "utf-8"))
```

Comment:
Not sure if we should encode the `Response` content with custom encoding. It may be incompatible with the underlying `StarletteResponse`

```python
# models.py
class Response:
    async def __call__(self, scope, receive, send):
        # `await body` returned content encoded by Response.encoding formerly
        body, headers = await self.body
        ...
        if self._stream is not None:
            response_cls = StarletteStreamingResponse
        else:
            response_cls = StarletteResponse
        # And the encoded body is passed to initialize StarletteResponse
        response = response_cls(body, status_code=self.status_code, headers=headers)
        self._prepare_cookies(response)
        ...
```

By checking `StarletteResponse`

```python
class Response:
    media_type = None
    charset = "utf-8"

    def __init__(...) -> None:
        ...
        self.body = self.render(content)

    def render(self, content: typing.Any) -> bytes:
        if content is None:
            return b""
        if isinstance(content, bytes):
            return content
        return content.encode(self.charset)
```

You see, the `StarletteResponse.render()` assume the content is bytes, otherwise it will encode the content with `utf-8`.

This is the reason I choose to encode text with `StarletteResponse.charset`.